airflow/scripts/git_scripts/fetch_repo_data.py

```diff
@@ -0,0 +1,58 @@
+import requests
+
+
+token = 'put_token_here'   # add your token
+
+
+def get_repo_data(owner, repo, token):
+
+    """ Fetch the data of the repository including stars, forks, watchers,
+     views and unique views(how many different users). views only work on repos you own"""
+
+    headers = {'Authorization': f'token {token}', 'Accept': 'application/vnd.github.v3+json'}
+
+    # Get repository information
+    repo_url = f"https://api.github.com/repos/{owner}/{repo}"
+    repo_response = requests.get(repo_url, headers=headers)
+    repo_data = repo_response.json()
+
+    stars = repo_data.get('stargazers_count', 0)
+    forks = repo_data.get('forks_count', 0)
+    watchers = repo_data.get('subscribers_count', 0)
+
+    # Get repository views
+    views_url = f"https://api.github.com/repos/{owner}/{repo}/traffic/views"
+    views_response = requests.get(views_url, headers=headers)
+    views_data = views_response.json()
+
+    views_count = views_data.get('count', 0)
+    unique_views = views_data.get('uniques', 0)
+
+    return {
+        'stars': stars,
+        'forks': forks,
+        'watchers': watchers,
+        'views_count': views_count,
+        'unique_views': unique_views
+    }
+
+def fetch_multiple_repos(repo_list, token):
```

Comment:
Use a generator here instead of append because you already using another for loop to get the data out.